Update tools appendix (#661)
OWASP/wstg: document/6-Appendix/A-Testing_Tools_Resource.md

## General Web Testing

### Web Proxies

- [OWASP ZAP](https://www.zaproxy.org)
  - The Zed Attack Proxy (ZAP) is an easy to use integrated penetration testing tool for finding vulnerabilities in web applications. It is designed to be used by people with a wide range of security experience and as such is ideal for developers and functional testers who are new to penetration testing.
  - ZAP provides automated scanners as well as a set of tools that allow you to find security vulnerabilities manually.

> - [Burpsuite Free](https://www.portswigger.net/Burp/)

Reviewer: ```suggestion
- [Burp Suite Community Edition](https://portswigger.net/burp/communitydownload)
```

>   - Burp Proxy is an intercepting proxy server for security testing of web applications it allows intercepting and modifying all HTTP(S) traffic passing in both directions, it can work with custom SSL certificates and non-proxy-aware clients.

Reviewer: ```suggestion
  - Burp Suite is an intercepting proxy for security testing. It allows intercepting and modifying all HTTP(S) traffic passing in both directions, it can work with custom SSL certificates and non-proxy-aware clients.
```

  - Request Maker is a tool for penetration testing. With it you can easily capture requests made by web pages, tamper with the URL, headers and POST data and, of course, make new requests
- [Cookie Editor](https://chrome.google.com/webstore/detail/fngmhnnpilhplaeedifhccceomclgfbg?hl=en-US)
  - Edit This Cookie is a cookie manager. You can add, delete, edit, search, protect and block cookies

### Testing for Specific Vulnerabilities

#### Testing for DOM-based XSS

- [BlueClosure BC Detect](https://www.blueclosure.com)

#### Testing for SQL Injection

- [sqlmap](http://sqlmap.org)

#### Testing SSL

- [OWASP O-Saft](https://owasp.org/www-project-o-saft/)
- [sslyze](https://github.com/nabla-c0d3/sslyze)
- [testssl.sh](https://github.com/drwetter/testssl.sh)
- [SSLScan](https://github.com/rbsec/sslscan)
- [SSLLabs](https://www.ssllabs.com/ssltest/)

#### Testing for Brute Force Attacks

##### Hash Crackers

- [John the Ripper](https://github.com/openwall/john)
- [hashcat](https://hashcat.net/hashcat/)

##### Remote Brute Force

- [Patator](https://github.com/lanjelot/patator)
- [THC Hydra](https://github.com/vanhauser-thc/thc-hydra)

> - [Burpsuite Professional (Intruder)](https://portswigger.net/burp)

Reviewer: ```suggestion
- [Burp Suite Professional (Intruder)](https://portswigger.net/burp/pro)
```

Reviewer: Let's remove this. Or we can link to the community edition. Intruder can be used, but it is throttled.

#### Fuzzers

- [Wfuzz](https://github.com/xmendez/wfuzz)

#### Google Hacking

> - [Google Hacking Diggity Project](https://resources.bishopfox.com/resources/tools/google-hacking-diggity/)

Reviewer: This is safe to be removed as Bishop Fox declared it obsolete and recommend theHarvester for similar needs.

Author: It looks like there's still a lot of useful information there - even if the tools isn't updated any more, is the information still useful?

Reviewer: I saw the information to be revolving around the tool. I'd refer to other books or PDFs that explain this better instead if we were to do that. It's no longer a "tool" though, so not part of this appendix.

  - Use Tamper Data to view and modify HTTP/HTTPS headers and post parameters
- [Firefox Web Developer](https://addons.mozilla.org/en-US/firefox/addon/web-developer/)
  - The Web Developer extension adds various web developer tools to the browser.

### Chrome Extensions

- [Chrome Web Developer](https://chrome.google.com/webstore/detail/bfbameneiokkgbdmiekhjnmfkcnldhhm)
  - The Web Developer extension adds a toolbar button to the browser with various web developer tools. This is the official port of the Web Developer extension for Chrome.
- [HTTP Request Maker](https://chrome.google.com/webstore/detail/kajfghlhfkcocafkcjlajldicbikpgnp?hl=en-US)
  - Request Maker is a tool for penetration testing. With it you can easily capture requests made by web pages, tamper with the URL, headers and POST data and, of course, make new requests
- [Cookie Editor](https://chrome.google.com/webstore/detail/fngmhnnpilhplaeedifhccceomclgfbg?hl=en-US)
  - Edit This Cookie is a cookie manager. You can add, delete, edit, search, protect and block cookies

### Testing for Specific Vulnerabilities

#### Testing for DOM-based XSS

> - [BlueClosure BC Detect](https://www.blueclosure.com)

Author: Never heard of this

Reviewer: Let's replace with [XSS Strike](https://github.com/s0md3v/XSStrike)

Author: That project doesn't seem to have had any commits in the last 12 months - is it still alive?

Reviewer: Interesting. Didn't see that to be outdated. Let's ignore that, we can remove this section.

#### Testing for SQL Injection

- [sqlmap](http://sqlmap.org)

#### Testing Oracle Vulnerabilities

> - [Toad for Oracle](https://www.quest.com/toad)

Author: Never heard of this, and not sure what it does. Doesn't really seem very webapp focused - remove?

Reviewer 2: Im familiar with Toad it's a sql*net client, but I don't think it's necessary to keep here.

Drop it.

- [TestSSLServer](https://www.bolet.org/TestSSLServer/)
- [SSLScan](https://github.com/rbsec/sslscan)
- [SSLLabs](https://www.ssllabs.com/ssltest/)

#### Testing for Brute Force Attacks

##### Hash Crackers

- [John the Ripper](https://github.com/openwall/john)
- [hashcat](https://hashcat.net/hashcat/)

##### Remote Brute Force

- [Patator](https://github.com/lanjelot/patator)
- [THC Hydra](https://github.com/vanhauser-thc/thc-hydra)
- [Burpsuite Professional (Intruder)](https://portswigger.net/burp)

#### Fuzzers

- [Wfuzz](https://github.com/xmendez/wfuzz)

#### Google Hacking

- [Google Hacking Diggity Project](https://resources.bishopfox.com/resources/tools/google-hacking-diggity/)
- [Google Hacking database](https://www.exploit-db.com/google-hacking-database/)

#### Slow HTTP

- [Slowloris](https://github.com/gkbrk/slowloris)
- [slowhttptest](https://github.com/shekyan/slowhttptest)

### Site Mirroring

- [wget](https://www.gnu.org/software/wget/)
- [wget for windows](http://gnuwin32.sourceforge.net/packages/wget.htm)
- [curl](https://curl.haxx.se/)

## Vulnerability Scanners

### Open Source

- [w3af](https://w3af.org)
- [OWASP ZAP](https://www.zaproxy.org)

### Commercial

> - [NGS Typhon](https://www.nccgroup.trust/uk/our-services/cyber-security/technology-solutions/)

Author: This just seems to be an advert for a pentesting company - I don't see any tool here?

Reviewer 2: Drop it, don't think the product exists anymore. A quick Google search shows lots of related early 2000s content.

- [HCL AppScan](https://www.hcltechsw.com/products/appscan)
- [Burpsuite Professional](https://portswigger.net/burp)
- [Acunetix Web Vulnerability Scanner](https://www.acunetix.com)

> - [MaxPatrol Security Scanner](https://www.ptsecurity.com/ww-en/products/maxpatrol/)

Author: Never heard of this

Reviewer 2: Seems to be a compluance/audit tool. Was able to find up-to-date Gartner mentions of it.

Author: Commercial, so has been removed.

> - [Parasoft SOAtest (more QA-type tool)](https://www.parasoft.com/products/soatest)

Author: Doesn't seem to be a working site

Reviewer 2: https://www.parasoft.com/products/parasoft-soatest/

> - [IndusFace WAS](https://www.indusface.com/web-application-scanning.php)

Author: Never heard of this

Reviewer 2: I'm ok with dropping it.

> 

Author: Are these relevant?

Also never heard of most of the code review tools above (but can't comment due to GitHub UI being a bit useless)

Reviewer 2: You mean Accepance Testing Tools?

If BDD was tyye only entry I'm good with it being dropped.

Acceptance testing tools are used to validate the functionality of web applications. Some follow a scripted approach and typically make use of a Unit Testing framework to construct test suites and test cases. Most, if not all, can be adapted to perform security specific tests in addition to functional tests.

### Open Source Tools

- [HtmlUnit](http://htmlunit.sourceforge.net)
  - A Java and JUnit based framework that uses the Apache HttpClient as the transport.
  - Very robust and configurable and is used as the engine for a number of other testing tools.

> - [Selenium](https://www.seleniumhq.org)

Reviewer 2: Why drop the slash?

https://www.selenium.dev/ seems to be the primary now.

Author: I gto rid of the slash because it's redundant (and not included for most of the other URLs) - I've updated it to the new .dev site
